pulpcore/tests/functional/api/test_tasks.py

```diff
@@ -134,14 +140,71 @@ def test_02_search_by_invalid_name(self):
 
     @skip_if(bool, 'task', False)
     def test_03_delete_tasks(self):
-        """Delete a task."""
-        # If this assertion fails, then either Pulp's tasking system or Pulp
+        """Delete created tasks."""
+        # If these assertions fail, then either Pulp's tasking system or Pulp
         # Smash's code for interacting with the tasking system has a bug.
-        self.assertIn(self.task['state'], P3_TASK_END_STATES)
-        self.client.delete(self.task['_href'])
-        with self.assertRaises(HTTPError):
-            self.client.get(self.task['_href'])
+        tasks = self.client.get(TASKS_PATH)
+        for task in tasks['results']:
+            self.assertIn(task['state'], P3_TASK_END_STATES)
+            self.client.delete(task['_href'])
+            with self.assertRaises(HTTPError):
+                self.client.get(task['_href'])
 
     def filter_tasks(self, criteria):
         """Filter tasks based on the provided criteria."""
         return self.client.get(TASKS_PATH, params=criteria)
+
+
+class FilterTaskCreatedResourcesTestCase(unittest.TestCase):
+    """Perform filtering over task resources.
+
+    This test targets the following issue:
+
+    * `Pulp #5180 <https://pulp.plan.io/issues/5180>`_
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variables."""
+        cls.client = api.Client(config.get_config(), api.json_handler)
+        cls.task = {}
+
+    def test_01_create_task_with_reserved_resources(self):
+        """Create a task that will create a new distribution."""
+        distribution_path = '{}file/file/'.format(BASE_DISTRIBUTION_PATH)
+        response = self.client.post(distribution_path, gen_distribution())
+
+        task = self.client.get(response['task'])
+        self.task.update(task)
+
+    @skip_if(bool, 'task', False)
+    def test_02_read_fields_created_resources_only(self):
+        """Read created resources from the requested fields."""
+        task = self.client.get(
+            self.task['_href'],
```

Comment:
Is it the goal to filter in the general task path or a specific task?
```suggestion
     TASKS_PATH,
```

Reply:
In order to trigger the method which was previously modified (https://github.com/pulp/pulpcore/pull/228), it is sufficient enough to create an HTTP GET request that targets a single task.

Reply:
Because of that, I do not need to delete all tasks in a database in the method setUpClass(). I do can just query the right one.